syncer/streamer_controller.go

```diff
@@ -120,10 +124,30 @@ type StreamerController struct {
 }
 
 // NewStreamerController creates a new streamer controller.
-func NewStreamerController(syncCfg replication.BinlogSyncerConfig, enableGTID bool, fromDB *dbconn.UpStreamConn, binlogType BinlogType, localBinlogDir string, timezone *time.Location) *StreamerController {
+func NewStreamerController(
+	syncCfg replication.BinlogSyncerConfig,
+	enableGTID bool,
+	fromDB *dbconn.UpStreamConn,
+	binlogType BinlogType,
+	localBinlogDir string,
+	timezone *time.Location,
+) *StreamerController {
+	var strategy retryStrategy = alwaysRetryStrategy{}
+	if binlogType != LocalBinlog {
+		strategy = &maxIntervalRetryStrategy{
+			interval: minErrorRetryInterval,
+		}
+	}
+	// let local binlog also return error to avoid infinity loop
+	failpoint.Inject("GetEventError", func() {
+		strategy = &maxIntervalRetryStrategy{
+			interval: minErrorRetryInterval,
+		}
+	})
 	streamerController := &StreamerController{
 		initBinlogType:    binlogType,
 		currentBinlogType: binlogType,
+		retryStrategy:     strategy,
```

Comment:
can we use func `NewStreamerController` to init streamerController here ?

Reply:
I don't get the point, this function is `NewStreamerController` 😕️